### DISCUSSION TITLE
Parse MSA segment using Terser

### DISCUSSION THREAD

**Discussion by jorge-zapata-medstone** [nHapiNET/nHapi#415] [Q&A] [answered]
## Description:
Hi, im using `Terser`  to parse a string message and when i reach the MSA segment it throws an error. 

Does anyone have any ideas on how to parse this segment?

Error message
```
Error: NHapi.Base.HL7Exception: Can't process message of version '2.5
MSA' - version not recognized
   at NHapi.Base.Parser.ParserBase.Parse(String message, ParserOptions parserOptions)
```

Heres my message
```
MSH|^~\\&|RAMSOFT|SENDING FACILITY|RAMSOFT|RECEIVING FACILITY|20101223202939-0400||ACK|101|P|2.5||||||||
MSA|AA|104|Request was Successfully processed
```

And a code fragment
```
_hl7Message = pipeParser.Parse(hl7Message.messageContent);
_hl7Terser = new Terser(_hl7Message);
```

**Comment by milkshakeuk** [comment, marked as answer]
Hi @jorge-zapata-medstone I can't replicate, I have created a dotnet fiddle showing it working [here](https://dotnetfiddle.net/8TTMHi).

```csharp
using System;
using NHapi.Base.Parser;
using NHapi.Base.Util;

var newline = '\r';
// adding new line into the string on linux, not needed on windows
var message = $"""
	MSH|^~\&|RAMSOFT|SENDING FACILITY|RAMSOFT|RECEIVING FACILITY|20101223202939-0400||ACK|101|P|2.5||||||||{newLine}
	MSA|AA|104|Request was Successfully processed{newLine}
	""";
		
var pipeParser = new PipeParser();
var parsed = pipeParser.Parse(message);
var terser = new Terser(parsed);

Console.WriteLine($"MSH.1: {terser.Get("/MSA-1")}");
Console.WriteLine($"MSH.2: {terser.Get("/MSA-2")}");
Console.WriteLine($"MSH.3: {terser.Get("/MSA-3")}");
```

I suspect it could be your message has the wrong line endings. In the HL7 specification, the end of each segment (line) should end with a carriage return i.e. `\r` (or `0x0D` in hexadecimal), if you're testing your code with just C# strings on Linux then the lines only end with a line feed i.e. `\n` (or `0x0A` in hexadecimal) as opposed to `\r\n` on Windows.

**Reply by jorge-zapata-medstone** [reply]
thanks, i just tested it again and it was the `\n` line ending
